src/spyglass/spikesorting/spikesorting_populator.py

```diff
@@ -59,199 +60,161 @@ def spikesorting_pipeline_populator(
-            }
-        ).fetch1("valid_times")[0]
+        interval_list = (IntervalList & interval_dict).fetch1("valid_times")[0]
+
         sort_interval_name = interval_list_name
         sort_interval = interval_list
+
         SortInterval.insert1(
             {
-                "nwb_file_name": nwb_file_name,
+                **nwbf_dict,
                 "sort_interval_name": sort_interval_name,
                 "sort_interval": sort_interval,
             },
             skip_duplicates=True,
         )
 
+    sort_dict = dict(**nwbf_dict, sort_interval_name=sort_interval_name)
+
+    # find desired sort group(s) for these settings
+    sort_group_id_list = (
+        (SortGroup.SortGroupElectrode * ElectrodeGroup)
+        & nwbf_dict
+        & probe_restriction
+    ).fetch("sort_group_id")
+
     # make spike sorting recording
     print("Generating spike sorting recording")
     for sort_group_id in sort_group_id_list:
         ssr_key = dict(
-            nwb_file_name=nwb_file_name,
+            **sort_dict,
             sort_group_id=sort_group_id,  # See SortGroup
-            sort_interval_name=sort_interval_name,  # First N seconds above
             preproc_params_name=preproc_params_name,  # See preproc_params
             interval_list_name=interval_list_name,
             team_name=team_name,
         )
         SpikeSortingRecordingSelection.insert1(ssr_key, skip_duplicates=True)
-    ssr_pj = (
-        SpikeSortingRecordingSelection()
-        & {
-            "nwb_file_name": nwb_file_name,
-            "interval_list_name": interval_list_name,
-        }
-    ).proj()
-    SpikeSortingRecording.populate([ssr_pj])
+
+    SpikeSortingRecording.populate(interval_dict)
 
     # Artifact detection
     print("Running artifact detection")
-    artifact_key_list = (ssr_pj).fetch("KEY")
-    for artifact_key in artifact_key_list:
-        artifact_key["artifact_params_name"] = artifact_parameters
-        ArtifactDetectionSelection().insert1(artifact_key, skip_duplicates=True)
-
-    art_pj = (
-        ArtifactDetectionSelection()
-        & {
-            "nwb_file_name": nwb_file_name,
-            "interval_list_name": interval_list_name,
-        }
-    ).proj()
-    ArtifactDetection.populate([art_pj])
+    artifact_keys = [
+        {**k, "artifact_params_name": artifact_parameters}
+        for k in (SpikeSortingRecordingSelection() & interval_dict).fetch("KEY")
+    ]
+    ArtifactDetectionSelection().insert(artifact_keys, skip_duplicates=True)
+    ArtifactDetection.populate(interval_dict)
 
     # Spike sorting
     print("Running spike sorting")
-    for artifact_key in artifact_key_list:
+    for artifact_key in artifact_keys:
         ss_key = dict(
-            **(ArtifactDetection & artifact_key).fetch1("KEY"),
-            **(ArtifactRemovedIntervalList() & artifact_key).fetch1("KEY"),
+            **(
+                ArtifactDetection * ArtifactRemovedIntervalList & artifact_key
```

Comment:
This join operation raises an error for me:

> DataJointError: Cannot join query expressions on dependent attribute `artifact_removed_valid_times`

I think it's because artifact_removed_valid_times is a secondary key in one table and a primary key in the other (?)

Reply:
Thanks for pointing this out @samuelbray32. I've reverted that commit